### PR TITLE
refactor message factory to reduce code duplication

### PR DESCRIPTION
**Pull request by junchaowu**
@blampe @abhinav @breerly

### CODE REVIEW

**Comment by breerly** [comment]
This looks logical - suggest landing to 0.19.0 branch

**Comment by junchaowu** [comment]
this branch will eventually land on the 0.19.0 branch. I am separate lots of change from https://github.com/uber/tchannel-python/pull/190 . so it is easy to review.

File: tchannel/tornado/message_factory.py [hunk 1]
```diff
@@ -428,3 +418,13 @@ def set_inbound_exception(self, protocol_error):
         reqres.argstreams[dst].set_exception(protocol_error)
 
         self.message_buffer.pop(protocol_error.id, None)
+
+
+def _find_incomplete_stream(reqres):
+    # find the incomplete stream
```
Comment (abhinav) [inline tchannel/tornado/message_factory.py]: Documentation shouldn't just repeat the function name.

**Comment by abhinav** [comment]
:+1: 

LGTM besides minor doc comment

**Pull request merged**